### DISCUSSION TITLE
Can I use esbuild plugins with Remix

### DISCUSSION THREAD

**Discussion by m4saurabh** [remix-run/remix#1108] [Q&A] [answered]
Is there a way to use esbuild plugins with Remix I wanted to use remix with [vanilla-extract](https://vanilla-extract.style/). They have a [esbuild plugin](https://vanilla-extract.style/documentation/setup/#esbuild) but I don't know how to use it with Remix.

**Comment by sergiodxa** [comment, marked as answer]
The esbuild config is not exposed so it’s not possible to change it, this is intentional so Remix is not tied to esbuild and can change it (they already changed from Rollup to esbuild once)

**Reply by m4saurabh** [reply]
Isn't it awkward that the underlying compiler is a subject to change.

**Reply by sergiodxa** [reply]
I don’t think so, you are using Remix as a compiler, if Remix uses another compiler internally it’s an implementation detail, they may eventually let you add Remix plugins (they may not) without they being related to the internal compiler they use.

**Reply by arvigeus** [reply]
We need another [craco](https://www.npmjs.com/package/@craco/craco). RCO? (I am half joking, of course)
